formats/json-tests/commonTest/src/kotlinx/serialization/InstantSerializationTest.kt

```diff
@@ -45,14 +45,21 @@ class InstantSerializationTest: JsonTestBase() {
             Pair(Instant.fromEpochSeconds(987654321, 123456789),
                 "{\"epochSeconds\":987654321,\"nanosecondsOfSecond\":123456789}"),
             Pair(Instant.fromEpochSeconds(987654321, 0),
-                "{\"epochSeconds\":987654321}"),
+                "{\"epochSeconds\":987654321,\"nanosecondsOfSecond\":0}"),
         )) {
             assertJsonFormAndRestored(serializer, instant, json)
         }
-        // check that having a `"nanosecondsOfSecond": 0` field doesn't break deserialization
+        // by default, `nanosecondsOfSecond` is optional
+        assertJsonFormAndRestored(serializer, Instant.fromEpochSeconds(987654321, 0),
+            "{\"epochSeconds\":987654321}", Json { })
```

Comment:
The comment might clarify that the default test json instance has encoding defaults enabled. the `nanosecondsOfSecond` element is always optional, but the format decides whether to serialize the default value. You might replace `Json {}` with `Json(default) { encodeDefaults = false }` to both take in the remainder of the default test configuration and communicate that it set to false here.